disable CUDA in llvm

opadron opened the pull request:
Adds a new variant to llvm: `cuda`, which if disabled (default), will set additional CMake variables so as to explicitly disable CUDA features from the llvm build.

These changes address an issue where a local cuda install outside of spack can leak into and break the build.  The variant doesn't add any dependencies on the `cuda` package, but perhaps it should?  Open to suggestions.

chuckatkins commented on var/spack/repos/builtin/packages/llvm/package.py:
```diff
@@ -588,6 +590,12 @@ def cmake_args(self):
+                '-DCUDA_TOOLKIT_ROOT_DIR:PATH=IGNORE',
+                '-DCUDA_SDK_ROOT_DIR:PATH=IGNORE',
+                '-DCUDA_NVCC_EXECUTABLE:FILEPATH=IGNORE'])
+
```
You'll also want to add `-DLIBOMPTARGET_DEP_CUDA_DRIVER_LIBRARIES=IGNORE`

opadron replied:
Thanks!  Added.

chuckatkins commented:
While this adds the cuda variant and seems to do the right thing when it's not enabled, it's not doing anything when it is enabled.  If there is going to be a separate cuda variant then it should add cuda as a dependency when enabled and set the paths accordingly.

chuckatkins commented:
> I was hoping to keep these changes limited to the minimum we need to work around our initial issue (that of the leaking CUDA install)

I agree with that, but since all this wants to accomplish is to disable cuda when it's not explicitly specified then I think the variant should be removed and the extra options added always.  A follow up would be to add the cuda variant and conditionally enable or disable the cuda options with the appropriate dependency information from the spack cuda package.

chuckatkins commented:
lgtm, up to the package maintainers now

opadron commented:
@chuckatkins PTAL.  I removed the variant and added notes at the two source locations that should clearly mark where it can be reintroduced.  I'm ready to prepare that follow-up issue once this LGTY.

chuckatkins commented:
@alalazo @adamjstewart ping?

This is currently blocking progress on #10482
